pkg/autodiscover/autodiscover.go

```diff
@@ -280,3 +290,81 @@ func getOpenshiftVersion(oClient clientconfigv1.ConfigV1Interface) (ver string,
 
 	return "", errors.New("could not get openshift version from clusterOperator")
 }
+
+// Get a map of csvs with its managed pods from the target namespaces
+func getOperatorCsvPods(csvList []*olmv1Alpha.ClusterServiceVersion) (map[string][]*corev1.Pod, error) {
+	client := clientsholder.GetClientsHolder()
+	csvToPodsMapping := make(map[string][]*corev1.Pod)
+
+	for _, csv := range csvList {
+		// Get target namespaces
+		operatorTargetNamespaces, err := getOperatorTargetNamespaces(csv, client)
+		if err != nil {
+			return csvToPodsMapping, err
+		}
+		var podList []*corev1.Pod
+
+		for _, targetNamespace := range operatorTargetNamespaces {
+			pods, err := getCsvPodsFromTargetNamespace(csv, strings.TrimSpace(targetNamespace), client)
+			if err != nil {
+				continue
+			}
+			podList = append(podList, pods...)
+		}
+
+		csvToPodsMapping[fmt.Sprintf("%s, namespace=%s", csv.Name, csv.Namespace)] = podList
+	}
+	return csvToPodsMapping, nil
+}
+
+// This function gets the pods of the specified csv in a target namespace
+func getCsvPodsFromTargetNamespace(csv *olmv1Alpha.ClusterServiceVersion, targetNamespace string, client *clientsholder.ClientsHolder) (managedPods []*corev1.Pod, err error) {
+	// Get all pods from the target namespace
+	podsList, err := client.K8sClient.CoreV1().Pods(targetNamespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for index := range podsList.Items {
+		labels := podsList.Items[index].Labels
+
+		if _, ok := labels["olm.managed"]; !ok { // Ignore non-OLM pods in the namespace
```

Comment:
not all pods are labeled, remove this line to capture all operator pods.

Reply:
I think this condition is necessary. I checked that OLM adds this annotation in the controller created pods. This filter condition is added to remove some other pods deployed in the same namespace.

Reply:
I checked with the hazelcast operator controller pod that we use and it didn't have the label, yet the pod traces back to the csv with the owner references, so would populate the CSVToPodListMap.

Reply:
Wait .. we are not considering the operator controller pod as the workload. Rather the pods created by the controller code. I checked with CNF operator and its created pods by the CR and we get this label.